relm4/src/app.rs

```diff
@@ -169,6 +170,55 @@ impl<M: Debug + 'static> RelmApp<M> {
         glib::MainContext::ref_thread_default().iteration(true);
     }
 
+    /// Runs the application, returns once the application is closed.
+    pub fn run_application<C>(self, payload: C::Init)
+    where
+        C: Component<Input = M>,
+        C::Root: AsRef<gtk::Application>,
+    {
+        let Self {
+            app,
+            broker,
+            args,
+            ..
+        } = self;
+
+        let hold_guard = Rc::new(RefCell::new(Some(app.hold())));
+        app.connect_window_added(move |_app, _window| {
+            hold_guard.borrow_mut().take();
+        });
```

Comment:
I think `Rc` is not necessary here and `Cell` should be used instead of `RefCell`. Same for the async version later.